uint32 id
float64 minSpeed
uint32 currLocation
uint8 status